### DISCUSSION TITLE
Recommendations for State Machine libraries in Java that are XState compatible?

### DISCUSSION THREAD

**Discussion by mattmazzola** [statelyai/xstate#3290] [General]
I have used XState in the past with TypeScript and really like the experience. The ability to define the entire machine declaratively and visualize in the browser before consuming in your application was nice.

I am going to be building new feature in a Java based project and wondered if there are any recommended libraries that are XState compatible. By this I mean I could take the provide json used by XState and give it to this library and it would function the same.

If there isn't a compatible library what is the closest?

**Comment by goldenratio** [comment]
+1 bumping this.
Any help is appreciated

**Reply by davidkpiano** [reply]
It's a good question, but I'm not familiar enough with the Java landscape to know the answer to this.

Someone would need to build an SCXML interpreter in Java.

**Comment by philippnagel** [comment]
@mattmazzola were you able to find something? Asking ourselves the same question.

**Reply by mattmazzola** [reply]
No, I was considering building our own solution since I only needed the minimal functionality for our use case. The most important being I can load the same JSON schema. This means I can leverage existing tooling for typescript, but use in Java. There are some good videos on recreating basics. 

However, I ended up shifting to work on a different project and did not have the time.
If I do something else with Java again, I might consider, but no plans as of now.

Good luck!
